index.html

```diff
@@ -1173,12 +1173,24 @@ <h3>Verification Material</h3>
             </p>
 
             <p>
+Implementers are often exposed to many different verification material formats
+such as
+<a href="https://en.wikipedia.org/wiki/Secure_Shell">SSH public keys</a>,
+<a href="https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail">
+PEM-encoded public keys</a>,
+hex-encoded public keys,
+<a data-cite="?RFC7517">JSON Web Keys</a>,
+<a href="#Multikey">Multikeys</a>, and
+<a data-cite="?8152#section-7">COSE public keys</a>.
 To increase the likelihood of interoperable implementations, this specification
-limits the number of formats for expressing verification material in a
-[=controller document=]. The fewer formats that implementers have to
-choose from, the more likely that interoperability will be achieved. This
-approach attempts to strike a delicate balance between easing implementation
-and providing support for formats that have historically had broad deployment.
+suggests limiting the number of formats used to express verification material in
+a [=controller document=]. The fewer formats that implementers have to choose
+from, the more likely that interoperability will be achieved for a particular
+ecosystem. Ideally, there would be one verification material format per
+application; however, history has shown both that multiple verification
+material formats might be used with any given application, and that
+implementers will support as many verification material formats as make
+sense for their ecosystem.
```

Comment:
```suggestion
limits the number of formats for expressing verification material in a
[=controller document=]. The fewer formats that implementers have to
choose from, the more likely that interoperability will be achieved. This
approach attempts to strike a delicate balance between easing implementation
and providing support for formats that have historically had broad deployment.
Implementations of this specification MUST NOT use key formats other than
those defined by this specification.
```

These changes address @msporny's request:
> Please provide concrete text that would address your objection.

Reply:
@selfissued , just to clarify the situation for the future. If, let us say in two years yet another widely adopted format comes to the fore, implementations should not use that if they want to be comformant, unless a new charter is accepted by the AC which explicitly accepts to add the new format to the list (noting that such a change would be a class 4 change, cannot be done by the WG under its current charter once the CD is published as a Rec).

I am not against that situation, giving the AC its say into a third format is probably a good thing. But I feel uncomfortable in the intermediate situation, when the new format is used (because it is used by a particular community) and the longish process to standardize its usage. Implementation would then not be comform to our spec.

All this is a longish reason why I am asking: would be o.k. using _SHOULD NOT_ instead of _MUST NOT_ in the last sentence?

Reply:
The reason it's "MUST NOT" is so there is a bright line between conformance and non-conformance to the specification - thereby increasing interoperability among implementations.

If people want to standardize other key formats and revise the specification to add them in the future, they can do so, but that's outside the scope of what this v1 specification is designed to accomplish.

Reply:
And I believe _SHOULD NOT_ achieves essentially the same: the meaning is that implemetations should take all necessary and possible actions to deviate from the standard and, I feel it is fine to allow for this in this particular case. There is a difference, after all, between _SHOULD NOT_ and _MAY NOT_.

We may agree to disagree on this point, let us see what others in the group say. I’m, otherwise, o.k. with your proposed text.

Reply:
+1 for SHOULD NOT.
-0.9 for MUST NOT.